File: 2022/aoc_2022/test/day_4_test.exs
defmodule Day4Test do
  use ExUnit.Case
  doctest Aoc2022.Day4

  test "part 1-example" do
    assert Aoc2022.Day4.solve("assets/day4_example.txt") == 2
  end

  test "part 1" do
    assert Aoc2022.Day4.solve("assets/day4.txt") == 515
  end

  test "part 2-example" do
    assert Aoc2022.Day4.solve2("assets/day4_example.txt") == 4
  end

  test "part 2" do
    assert Aoc2022.Day4.solve2("assets/day4.txt") == 883
  end

end

File: 2024/aoc2024.Tests/Day4Test.cs
using System;
using System.IO;
using aoc2024;
using day_2;
using JetBrains.Annotations;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace aoc2024.Tests;

[TestClass]
[TestSubject(typeof(Day4))]
public class Day4Test
{

    [TestMethod]
    public void testSolve1Example()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day4-example.txt");
        Assert.AreEqual(18, Day4.solve1(filePath));
    }
    [TestMethod]
    public void testSolve1()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day4-input.txt");
        Assert.AreEqual(2562, Day4.solve1(filePath));
    }
    [TestMethod]
    public void testSolve2Example()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day4-2-example.txt");
        Assert.AreEqual(9, Day4.solve2(filePath));
    }
    [TestMethod]
    public void testSolve2()
    {
        var projectDirectory = Directory.GetParent(AppContext.BaseDirectory).FullName;
        var filePath = Path.Combine(projectDirectory, "inputs", "day4-input.txt");
        Assert.AreEqual(1902, Day4.solve2(filePath));
    }
}